### PR TITLE
Update QD and documentation.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says.

### CODE REVIEW

File: rmw_dds_common/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -95,7 +95,17 @@ The tests aim to cover both typical usage and corner cases, but are quantified b
...
+
+This includes:
+
+- tracking and reporting line coverage statistics
+- achieving and maintaining a reasonable branch line coverage (90-100%)
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed [here](https://ci.ros2.org/job/ci_linux_coverage/lastSuccessfulBuild/cobertura/src_ros2_rmw_dds_common_rmw_dds_common_include_rmw_dds_common/) and [here](https://ci.ros2.org/job/ci_linux_coverage/lastSuccessfulBuild/cobertura/src_ros2_rmw_dds_common_rmw_dds_common_src/). This package does not yet meet the 95% coverage guideline, but it is currently above 90%.
```
Comment (brawner) [inline rmw_dds_common/QUALITY_DECLARATION.md:109]: Add link to on-boarding docs to describe how these statistics are calculated.
https://index.ros.org/doc/ros2/Contributing/ROS-2-On-boarding-Guide/#note-on-coverage-runs
